Firmware/MotorControl/controller.cpp

```diff
@@ -316,36 +316,35 @@ bool Controller::update(float* current_setpoint_output) {
             set_error(ERROR_INVALID_ESTIMATE);
             return false;
         }
-        Iq = limitVel(config_.vel_limit, *vel_estimate_src, vel_gain, Iq);
+        torque = limitVel(config_.vel_limit, *vel_estimate_src, vel_gain, torque);
     }
 
-    // Current limiting
-    // TODO: Change to controller working in torque units
-    // and get the torque limits from a function of the motor
+    // Limit max torque to a user defined torque limit. This functions as an acceleration limit.
+    // The motor object handles current limiting
     bool limited = false;
-    float Ilim = axis_->motor_.effective_current_lim();
-    if (Iq > Ilim) {
+    float Tlim = axis_->motor_.config_.torque_lim;
+    if (torque > Tlim) {
         limited = true;
-        Iq = Ilim;
+        torque = Tlim;
     }
-    if (Iq < -Ilim) {
+    if (torque < -Tlim) {
         limited = true;
-        Iq = -Ilim;
+        torque = -Tlim;
     }
 
     // Velocity integrator (behaviour dependent on limiting)
     if (config_.control_mode < CONTROL_MODE_VELOCITY_CONTROL) {
         // reset integral if not in use
-        vel_integrator_current_ = 0.0f;
+        vel_integrator_torque_ = 0.0f;
     } else {
         if (limited) {
             // TODO make decayfactor configurable
-            vel_integrator_current_ *= 0.99f;
+            vel_integrator_torque_ *= 0.99f;
         } else {
-            vel_integrator_current_ += ((vel_integrator_gain * gain_scheduling_multiplier) * current_meas_period) * v_err;
+            vel_integrator_torque_ += ((vel_integrator_gain * gain_scheduling_multiplier) * current_meas_period) * v_err;
         }
     }
```

Comment:
Since this no longer considers all kinds of current/torque limits (just the straight torque one), we are not implementing the anti-windup correctly. We need to consider all saturation effects on the "actuator" i.e. torque production.
I will follow up on Discord with some theory.